Taskfile.yml

```diff
@@ -346,22 +312,159 @@ tasks:
+    deps:
+      - deploy:kubernetes:setup-cluster
+    vars:
+      # Helm args
+      HELM_CHART_PATH:  '{{ .ROOT_DIR }}/install/charts/dir'
+      HELM_VALUES_PATH: '{{ .ROOT_DIR }}/install/charts/dir/values.yaml'
+    cmds:
+      # Generate private key if it doesn't exist
+      - |
+        test -f /tmp/node.privkey || openssl genpkey -algorithm ED25519 -out /tmp/node.privkey
+
+      # Generate the bootstrap peer ID and export it to the environment file
+      - |
+        bootstrap_peerid=$({{ .BIN_DIR }}/dirctl network info /tmp/node.privkey)
+        echo "PEER ID: ${bootstrap_peerid}"
+        echo BOOTSTRAP_PEER_ID="${bootstrap_peerid}" > .env
+
+      # Deploy the bootstrap server using Helm
+      - |
+        {{ .HELM_BIN }} dependency build {{ .HELM_CHART_PATH }}
+        {{ .HELM_BIN }} upgrade agntcy-dir \
+          {{ .HELM_CHART_PATH }} \
+          -f {{ .HELM_VALUES_PATH }} \
+          --set apiserver.image.tag="{{ .IMAGE_TAG }}" \
+          --set apiserver.privKey="$(cat /tmp/node.privkey)" \
+          --set apiserver.config.routing.key_path="/etc/agntcy/dir/node.privkey" \
+          --set apiserver.config.routing.listen_address="/ip4/0.0.0.0/tcp/8999" \
+          --set apiserver.config.oci.registry_address="agntcy-dir-zot.bootstrap.svc.cluster.local:5000" \
+          --namespace "bootstrap" \
+          --create-namespace \
+          --install \
+          --wait \
+          --wait-for-jobs \
+          --timeout "15m"
+
+  deploy:kubernetes:network:
+    aliases: [deploy:network]
+    desc: Deploy a network of Directory servers in Kubernetes (1 bootstrap + 3 peers)
+    deps:
+      - deploy:kubernetes:network:bootstrap
+    vars:
+      HELM_CHART_PATH:  '{{ .ROOT_DIR }}/install/charts/dir'
+      HELM_VALUES_PATH: '{{ .ROOT_DIR }}/install/charts/dir/values.yaml'
+    cmds:
+      # Deploy the peer servers using Helm
+      - for:
+          matrix:
+            PEER: ['peer1', 'peer2', 'peer3']
+        cmd: |
+          export $(cat .env)
+          {{ .HELM_BIN }} upgrade agntcy-dir \
+            {{ .HELM_CHART_PATH }} \
+            -f {{ .HELM_VALUES_PATH }} \
+            --set apiserver.image.tag="{{ .IMAGE_TAG }}" \
+            --set apiserver.config.oci.registry_address="agntcy-dir-zot.{{ .ITEM.PEER }}.svc.cluster.local:5000" \
+            --set apiserver.config.routing.bootstrap_peers[0]="/dns4/agntcy-dir-apiserver-routing.bootstrap.svc.cluster.local/tcp/8999/p2p/${BOOTSTRAP_PEER_ID}" \
+            --namespace "{{ .ITEM.PEER }}" \
+            --create-namespace \
+            --install \
+            --wait \
+            --wait-for-jobs \
+            --timeout "15m"
+
+      # Set up port-forwarding for the peers
+      - kubectl port-forward svc/agntcy-dir-apiserver -n peer1 8890:8888 &
+      - kubectl port-forward svc/agntcy-dir-apiserver -n peer2 8891:8888 &
+      - kubectl port-forward svc/agntcy-dir-apiserver -n peer3 8892:8888 &
+      - sleep 1
```

Comment:
Maybe it would be better to create a task for port-forwards, and use it before tests along with defer teardown task of the port-forward.